src/pages/api/semaphore-grant.ts

```diff
@@ -0,0 +1,123 @@
+    twitter: Twitter__c,
+    telegram: Alternative_Contact__c,
+    repeatApplicant: Repeat_Applicant__c,
+    canTheEFReachOut: Can_the_EF_reach_out__c,
+    additionalInfo: Additional_Information__c
+  } = body;
+  const { SF_PROD_LOGIN_URL, SF_PROD_USERNAME, SF_PROD_PASSWORD, SF_PROD_SECURITY_TOKEN } =
+    process.env;
+
+  const conn = new jsforce.Connection({
+    // you can change loginUrl to connect to sandbox or prerelease env.
+    loginUrl: SF_PROD_LOGIN_URL
+  });
+
+  conn.login(SF_PROD_USERNAME!, `${SF_PROD_PASSWORD}${SF_PROD_SECURITY_TOKEN}`, err => {
+    if (err) {
+      return console.error(err);
+    }
+
+    const application = {
+      FirstName: FirstName.trim(),
+      LastName: LastName.trim(),
+      Email: Email.trim(),
+      Company: Company.trim(),
+      Applying_as_a__c: Applying_as_a__c.trim(),
+      Applying_as_Other__c: Applying_as_Other__c.trim(),
+      npsp__CompanyCountry__c: npsp__CompanyCountry__c.trim(),
+      Countries_of_Team__c: Countries_of_Team__c.trim(),
+      Time_Zone__c: Time_Zone__c.trim(),
+      Project_Name__c: Project_Name__c.trim(),
+      Project_Description__c: Project_Description__c.trim(),
+      Category__c: Category__c.trim(),
+      Team_Profile__c: Team_Profile__c.trim(),
+      Previous_Work__c: Previous_Work__c.trim(),
+      Grant_Scope__c: Grant_Scope__c.trim(),
+      Impact__c: Impact__c.trim(),
+      Problem_Being_Solved__c: Problem_Being_Solved__c.trim(),
+      Is_it_a_Public_Good__c: Is_it_a_Public_Good__c.trim(),
+      Requested_Amount__c: Requested_Amount__c.trim(),
+      Proposed_Timeline__c: Proposed_Timeline__c.trim(),
+      Challenges__c: Challenges__c.trim(),
+      Additional_support_requests__c: Additional_support_requests__c.trim(),
+      Referral_Source__c: Referral_Source__c.trim(),
+      Referral_Source_if_Other__c: Referral_Source_if_Other__c.trim(),
+      Would_you_share_your_research__c: Would_you_share_your_research__c.trim(),
+      LinkedIn_Profile__c: LinkedIn_Profile__c.trim(),
+      Twitter__c: Twitter__c.trim(),
+      Alternative_Contact__c: Alternative_Contact__c.trim(),
+      Repeat_Applicant__c: Repeat_Applicant__c,
+      Can_the_EF_reach_out__c: Can_the_EF_reach_out__c,
+      Additional_Information__c: Additional_Information__c.trim(),
+
+      Proactive_Community_Grants_Round__c: 'Semaphore Community Grants 2022', // this value is hardwired, depending on the type of grant round
+      RecordTypeId: process.env.SF_RECORD_TYPE_GRANTS_ROUND!
+    };
+
+    // Single record creation
+    conn.sobject('Lead').create(application, async (err, ret) => {
+      if (err || !ret.success) {
+        console.error(err);
+        res.status(400).json({ status: 'fail' });
+        return;
+      }
+
+      // send submission data to a google spreadsheet
+      try {
+        await addRowToSpreadsheet(
+          {
+            id: '1YpA7U94D9t7wV4FabRbKJLB8GHhGpszqukhozhqLplE',
+            sheetName: 'Applications'
```

Comment:
I had to hardcode these values since Netlify is giving me the following error:
```
Your environment variables exceed the 4KB limit imposed by AWS Lambda. Please consider reducing them.
```
Those vars are not sensitive information since you need the proper credentials to access that doc.

Perhaps we could do the same for the other lambda functions to avoid having this error in the future.

cc @samajammin

Reply:
Ya I support reviewing our ENV variables to see which must be private. Spreadsheet ID & name don't seem necessary.